src/authentification/components/AuthRegisterForm.tsx

```diff
@@ -1,26 +1,42 @@
 import { useForm, FormProvider } from "react-hook-form";
 import { useRecoilState } from "recoil";
 import { User } from "../types/userType";
-import { userState } from "../state/userState";
+import { userState, loggedState } from "../state/userState";
 import {
   usernameMinLength,
   usernameMaxLength,
+  passwordMinLength,
+  passwordMaxLength,
+  passwordRegexp,
 } from "../const/userInputRequirements";
-import EmailInput from "./inputs/EmailInput";
-import PasswordInput from "./inputs/PasswordInput";
-import { createUserWithEmailAndPassword } from "firebase/auth";
+import { createUserWithEmailAndPassword, updateProfile } from "firebase/auth";
 import { auth } from "../../firebase/firebase";
 import { useNavigate } from "react-router-dom";
+import InputField from "./inputs/InputField";
 
 export default function AuthRegisterForm() {
   const navigate = useNavigate();
 
   const form = useForm<User>();
   const [userData, setUserData] = useRecoilState(userState);
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loggedState);
 
   const onSubmit = (data: User) => {
     createUserWithEmailAndPassword(auth, data.email, data.password)
-      .then(() => {
+      .then(async (userCredential) => {
+        if (!userCredential || !auth.currentUser)
```

Comment:
If the if statement goes in the second row, than put brackets for better readability.
```
if (!userCredential || !auth.currentUser){
      throw new Error("User data could not be found.");
}
```

This is good:
`if(something) return null;
`

Reply:
Added brackets.